tests/serac_thermal_structural_solver.cpp

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2019-2020, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include <gtest/gtest.h>
+
+#include <fstream>
+
+#include "coefficients/linear_coefficient.hpp"
+#include "mfem.hpp"
+#include "serac_config.hpp"
+#include "solvers/thermal_structural_solver.hpp"
+
+void   InitialDeformation(const mfem::Vector &x, mfem::Vector &y);
+void   InitialVelocity(const mfem::Vector &x, mfem::Vector &v);
+double InitialTemperature(const mfem::Vector &x);
+
+TEST(dynamic_solver, dyn_solve)
+{
+  MPI_Barrier(MPI_COMM_WORLD);
+
+  // mesh
+  std::string base_mesh_file = std::string(SERAC_SRC_DIR) + "/data/beam-hex.mesh";
+  const char *mesh_file      = base_mesh_file.c_str();
+
+  // Open the mesh
+  std::ifstream imesh(mesh_file);
+  auto          mesh = std::make_shared<mfem::Mesh>(imesh, 1, 1, true);
+  imesh.close();
+
+  mesh->UniformRefinement();
+
+  auto pmesh = std::make_shared<mfem::ParMesh>(MPI_COMM_WORLD, *mesh);
+
+  int dim = pmesh->Dimension();
+
+  // define a boundary attribute array and initialize to 0
+  std::vector<int> ess_bdr(pmesh->bdr_attributes.Max(), 0);
+
+  // boundary attribute 1 (index 0) is fixed (Dirichlet)
+  ess_bdr[0] = 1;
+
+  auto deform = std::make_shared<mfem::VectorFunctionCoefficient>(dim, InitialDeformation);
+  auto velo   = std::make_shared<mfem::VectorFunctionCoefficient>(dim, InitialVelocity);
+  auto temp   = std::make_shared<mfem::FunctionCoefficient>(InitialTemperature);
+  auto kappa = std::make_shared<mfem::ConstantCoefficient>(0.5);
```

Comment:
Since the function bodies of Initial***() functions are small, I think it's more readable to use lambdas / stdfunctionCoefficient to include them in the body of the test

Reply:
Agreed! I will fix this.